lib/trace-agent.js

```diff
@@ -202,7 +202,7 @@ TraceAgent.prototype.parseContextFromHeader = function(str) {
   }
   return {
     traceId: matches[1],
-    spanId: matches[2],
+    spanId: Number(matches[2]),
```

Comment:
When the argument cannot be converted to a number, `Number(argument)` returns `NaN`. I don't think the Trace API would like seeing `NaN`s here.